Inplace support for binary_ng ops

mcw-anasuya opened the pull request:
### Ticket
#16143

### Problem description
Provide inplace support for binary_ng op with a validation that checks if input a can hold the output tensor dimensions.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [ ] New/Existing tests provide coverage for changes

Commit:
#16153: Implement binary-ng fused input activations

mcw-anasuya committed:
Add inplace support for binary_ng

mcw-anasuya committed:
Support for sub_ and mul_

KalaivaniMCW commented on ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_device_operation.cpp:
```diff
@@ -76,6 +76,21 @@ void BinaryNgDeviceOperation::validate_on_program_cache_miss(
 
+    if (!tensor_args.output_tensor.has_value()) {  // Inplace operation
+        const auto& input_tensor_a = tensor_args.input_tensor_a;
+        const auto& output_shape = compute_output_specs(attributes, tensor_args).shape();
```
is there another function called compute_broadcasted shape ?

mcw-anasuya replied:
No. Broadcasted shape is being computed inside compute_output_specs.

tt-aho committed:
#0: Fixes for remote circular buffer synchronization

Co-authored-by: yugaoTT <yugao@tenstorrent.com>

SeanNijjar committed:
Add cluster-axis API support to reduce scatter (#16293)


- Also added additional cluster axis API tests for all-gather async

TT-billteng committed:
split ttnn unit tests 8 ways (#16382)

### Ticket
None

### Problem description
We're hitting the 45 minute timeout limit for some TTNN unit test jobs
because there's too many tests in some groups

### What's changed
Split TTNN unit tests into 8 groups from 6

TT-billteng committed:
split ttnn tests into 10 groups (#16383)

### Ticket
None

### Problem description
Seems like an 8-way split still leaves some test groups to take a very
long time

### What's changed
Split 10 ways

rayraykay committed:
Revert "#16356: Program Dispatch Modifications for MeshWorkload"

This reverts commit 3755428c7e5db16f11a0e7751c6277d0b5ffcb01.

tt-aho committed:
#0: [skip ci] Initial tech report for Sub-Device feature (#16387)

blozano-tt committed:
Adapt to tt-system-tools hugepages configuration (#14396)

### Tickets

#### metal-internal-workflows
Update provisioning to use new hugepages flow from tt-system-tools
-
https://github.com/tenstorrent-metal/metal-internal-workflows/issues/278
- https://github.com/tenstorrent-metal/metal-internal-workflows/pull/327

#### tt-metal
Many metal machines don't have correct hugepages config
https://github.com/tenstorrent/tt-metal/issues/15675

### Problem description
The old hugepages configuration was janky, and the plan of record is to
move to use tt-system-tools method.

### What's changed
- Remove invocation of `sudo /etc/rc.local` in mount weka script.
- Use our custom action `ensure-active-weka-mount` consistently
everywhere we try to mount weka
- Remove setup_hugepages.py
- Add forced garbage collection after every pytest execution to reduce
pressure on system memory

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [x] [Model regression CI testing
passes](https://github.com/tenstorrent/tt-metal/actions/runs/12290324750)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

---------

Co-authored-by: Raymond Kim <rkim@tenstorrent.com>

mcw-anasuya committed:
Support fused activations on inputs

mcw-anasuya commented:
Changes pertaining to this PR has been merged to main in #16449 
Hence marking this PR as closed.
